### DISCUSSION TITLE
Test swanctl.conf before reloading live process

### DISCUSSION THREAD

**Discussion by madalinignisca** [strongswan/strongswan#2247] [Q&A] [answered]
Sorry if I may have missed it from documentation, but I did search a bit.

Is there a way for swanctl to test a new version of swanctl.conf before actually reloading all required things?

My scenario is for automation, where I'd like to trigger a test, like for example how Nginx supports to use `-t /tmp/newNginx.conf` and will ensure me if the configuration files are OK, just before swanctl will trigger reload of configuration.

Is that also available for strongswan.conf?

**Comment by tobiasbrunner** [comment, marked as answer]
> Is there a way for swanctl to test a new version of swanctl.conf before actually reloading all required things?

No. The swanctl command can't really validate it as it basically only converts the strongswan.conf-style file to various vici messages. The _vici_ plugin then might reject individual messages because options etc. are invalid.

> My scenario is for automation, where I'd like to trigger a test, like for example how Nginx supports to use `-t /tmp/newNginx.conf` and will ensure me if the configuration files are OK, just before swanctl will trigger reload of configuration.

Why should they be invalid if you generate them? Also, you might want to consider using [vici](https://docs.strongswan.org/docs/5.9/plugins/vici.html) directly.

> Is that also available for strongswan.conf?

No, that file has no pre-defined options. So while some components or plugins might reject values read from it at runtime (not all options will actually have an effect when reloading the file), there is no central component that has any knowledge about the available options or their valid values.
